### ISSUE TITLE
Syntax doc should advertise and describe unicode identifiers

### ISSUE THREAD

**Issue by ronaldxs** [perl6/doc#1039]
The Syntax doc, [docs.perl6.org/language/syntax](https://docs.perl6.org/language/syntax), has a sentence about Unicode that says that Perl 6 is Unicode text but does seem to advertise applications of Unicode source, other than maybe the hyper operator.  I think it might be helpful for the description of identifiers to mention the use of alternative lettering/writing systems from Unicode as well as mathematical symbols as mentioned in a [Rosetta Code example](https://rosettacode.org/wiki/Unicode_variable_names#Perl_6).  It may also be useful to discuss the use of Unicode subscripts and superscripts and the exclusion of numeric Unicode subscripts and superscripts from identifiers (see discussions [IRC #perl6 2016-04-14](https://irclog.perlgeek.de/perl6/2016-04-14#i_12340102) and an [AlexDaniel gist](https://gist.github.com/AlexDaniel/c89bd2786f9b63f31e4c)).  My knowledge in this area is limited and there are likely to be other applications of Unicode source code that I have overlooked.

**Comment by AlexDaniel** [comment]
What about https://docs.perl6.org/language/unicode_texas ?

**Comment by ronaldxs** [comment]
See subsequent IRC discussion: [https://irclog.perlgeek.de/perl6/2016-12-07#i_13700922](https://irclog.perlgeek.de/perl6/2016-12-07#i_13700922).

**Comment by jonathanstowe** [comment]
```
class वर्ग { }; say वर्ग.new
```

**Comment by ronaldxs** [comment]
First cut at documentation update and opportunity for anyone who would like to comment before PR:

[https://github.com/ronaldxs/doc/commit/b7e4d5354cc1b4b2b99d59471044e72812af00ef](https://github.com/ronaldxs/doc/commit/b7e4d5354cc1b4b2b99d59471044e72812af00ef)

**Comment by ronaldxs** [comment]
A partial resolution was merged but leaving open ftm while I chase down some loose ends.

The roast tests @AlexDaniel mentioned related to subscripting, [S15/Unicode numbers.t](https://github.com/perl6/roast/blob/da6fde44ab97cdcbb50bf181528f55671914612c/S15-literals/numbers.t), suggest that at some point the [syntax doc int literals section](https://docs.perl6.org/language/syntax#Int_literals) is waiting for a unicode example. [Unicode numeric values](https://docs.perl6.org/language/unicode_texas#Numeric_values) are documented in Unicode vs Texas to include No(Number, other) property codepoints but seem to also include the Nl (Number, Letter) property.

 A [StackExchange question on P6 identifiers](http://stackoverflow.com/questions/34689850/whats-allowed-in-a-perl-6-identifier) notes that superscript exponents are not documented as [operators](https://docs.perl6.org/language/operators) even though they are included in [roast S32-num/power.t](https://github.com/perl6/roast/blob/master/S32-num/power.t).

**Comment by ronaldxs** [comment]
The Unicode vs Texas page has recently been updated to note that Nl property codepoints behave like No codepoints.  The superscript exponent operator documentation concern is now separate Issue #1098.  Waiting on feedback on Nl example to close.

**Comment by ronaldxs** [comment]
Closing for lack of further interest.

**Comment by AlexDaniel** [comment]
I don't think there is any problem with the interest. The thing is, we already mention unicode ops often enough (e.g. there is even a special page for it), and docs even use them sometimes.

Personally I think that docs should switch to non-texas ops fully (e.g. always ``…``, ``«»`` and curly quotes). However, I mentioned this once and faced a very strong opposition (something about newcomers being confused because of this, not sure if I agree that this is a problem). Not sure what's the point of “advertising” them if we don't use them 🤷‍.

**Comment by jonathanstowe** [comment]
The only problem I have with unicode operators is that I  don't know how to type half of them and some of those I do are actually more keystrokes than the ascii variants.

**Comment by AlexDaniel** [comment]
Well, but nobody is forcing you to improve your input methods? My suggestion regarding the docs is mostly about converting semi-automatically what we have already, and you can keep writing texas ops for the new stuff (which will be converted a bit later).

But people always try find reasons not to do this, so nevermind.

**Comment by finanalyst** [comment]
Jonathan stated the same reason I don't use unicode for my perl6 
programs - I have not found an efficient way to input them, though I 
have looked several times.

Perhaps someone could put together a page on an efficient input system 
for Ubuntu, Mac, and Windows - not a general page on how to input 
unicode but a specific one for each of the common codes Perl6 uses.

Since documentation is educational, and single unicode characters are 
preferred, perhaps the documentation should primarily use unicode. 
However, it is going to be quite some time before the generality of 
computer users (and most programmers start out as newbies) work with 
unicode. So at least on each page (perhaps as link on a footer or 
header) there is a reference to the documentation on these codes and the 
ASCII equivalents.


On Wednesday, January 04, 2017 04:35 AM, Aleks-Daniel 
Jakimenko-Aleksejev wrote:
